src/typescript-service.ts

```diff
@@ -338,6 +338,847 @@ export default class TypeScriptService {
 		});
 	}
 
+	getDocumentSymbol(params: DocumentSymbolParams): Promise<SymbolInformation[]> {
+		const uri = params.textDocument.uri;
+		return this.ensureFilesForHoverAndDefinition(uri).then(() => {
+			const fileName = util.uri2path(uri);
+			const config = this.projectManager.getConfiguration(uri);
+			return config.prepare(this.connection).then(() => {
+				this.projectManager.syncConfiguration(config, this.connection);
+				const sourceFile = config.program.getSourceFile(fileName);
+				const tree = config.service.getNavigationTree(fileName);
+				const limit = 100000;
+				const result: SymbolInformation[] = [];
+				this.flattenNavigationTreeItem(tree, null, sourceFile, result, limit);
```

Comment:
When tried to run in on Windows in non-strict mode (from vscode) I got the following items (extract):
```
  { name: 'TypeScriptService',
    kind: 15,
    location:
     { uri: 'file:///c:/projects/langserver-typescript/src/connection.ts',
       range: [Object] } },
  { name: 'util',
    kind: 15,
    location:
     { uri: 'file:///c:/projects/langserver-typescript/src/connection.ts',
       range: [Object] } },
```
Note absolute uris, we should make sure we are using relative ones